src/react/workflow/SourceBucketOption.tsx

```diff
@@ -17,9 +16,9 @@ import { useEffect } from 'react';
 type DisableOptionProps = {
   disableOption?: (obj: {
     isBucketVersionned?: boolean;
-    locationInfos?: LocationStorageInfos;
+    isReplicationSourceSupported?: boolean;
     isVeeamBucket: boolean;
-  }) => boolean;
+  }) => { disabled: boolean; disabledMessage?: JSX.Element };
```

Comment:
type `ReactNode` is better

Reply:
We only want a string or an ul element. Maybe we should use a union type ?